File: test-case/char/single/Single.kt
import soda.kotlin.unittest.*

class Solution {
    fun nextChar(ch: Char): Char = (ch + 1).toChar()
}

class Single : (String) -> String {
    override fun invoke(text: String): String {
        val work = GenericTestWork.create(Solution()::nextChar)
        // work.validator = (T, T) -> Boolean
        work.compareSerial = true
        return work.run(text)
    }

}

fun main() {
    print(Single()(Utils.fromStdin()))
}

File: test-case/char/single/Single.cs
using Soda.Unittest;

public class Solution {
    public char NextChar(char ch) {
        return (char) (ch + 1);
    }
}

public class Single
{
    public static void Main(string[] args)
    {
        var work = WorkFactory.Create(Utils.Fn(new Solution().NextChar));
        // work.SetValidator((e, r) => ...);
        work.CompareSerial = true;
        Console.WriteLine(work.Run(Utils.readStdin()));
    }
}
